### ISSUE TITLE
Explanation of LL

### ISSUE THREAD

**Issue by toddgauntgs** [MicrosoftDocs/Azure-IoT-C-SDK-Reference#22]
What does this naming convention mean? Low Level?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 7a34552c-e04a-c231-0210-68acb4e00c80
* Version Independent ID: d39f9460-cfb1-5065-17c5-505ad18ec946
* Content: [iothub_client_core_ll.h header file reference - Azure IoT C SDK](https://docs.microsoft.com/en-us/azure/iot-hub/iot-c-sdk-ref/iothub-client-core-ll-h)
* Content Source: [iothub-client-core-ll-h.md](https://github.com/MicrosoftDocs/Azure-IoT-C-SDK-Reference/blob/live/iothub-client-core-ll-h.md)
* Service: **iot-hub**
* GitHub Login: @wesmc7777
* Microsoft Alias: **wesmc**

**Comment by wesmc7777** [comment]
Hi @toddgauntgs, thanks for the question. 

LL stands for Lower layer.  You might read over [the iothub_client_core_ll.h reference here](https://docs.microsoft.com/en-us/azure/iot-hub/iot-c-sdk-ref/iothub-client-core-ll-h). The lower layer APIs give you more control over scheduling.

**Comment by toddgauntgs** [comment]
Thanks wesmc7777. The generated documentation seems a little sparse for these APIs, but I was able to figure out most of what I needed by reading through the code.

**Comment by wesmc7777** [comment]
@toddgauntgs I agree we need a lot more commenting in this SDK which will then be published in this reference. This is a new reference and we are driving initiatives with the team to add more helpful comments for the reference in the next few sprints. Making your voice heard helps us to know what kind of priority to put on this to help our customers.

**Comment by toddgauntgs** [comment]
@wesmc7777 I'm working on a developer guide for integrating this SDK with my company's certificate services, so if you guys are taking documentation contributions to the API I might be able to contribute back from the perspective of an SDK user.

**Comment by wesmc7777** [comment]
@toddgauntgs Yes we are very interested in documentation contributions and we want to make sure contributors like yourself receive full credit and accolades for contributions. We note this specifically here, [Contributing](https://github.com/Azure/azure-iot-sdk-c/blob/master/.github/CONTRIBUTING.md). 

At this time, this particular reference is generated entirely from code commenting in the [azure-iot-c-sdk](https://github.com/Azure/azure-iot-sdk-c). The bar for code quality and maintainability in the SDK might be a little higher than some would anticipate from a docs perspective but, we are certainly interested in accepting your contributions. We might just have to tranform some contributions a bit.

We might also want to consider some of your contributions in the appropriate [IoT "Service" documentation sets](https://docs.microsoft.com/en-us/azure/iot-fundamentals/) in [azure-docs](https://github.com/MicrosoftDocs/azure-docs) repository.

Before investing a large amount of time in a contribution, ping me so I can make sure your contribution will align with project goals and is not redundant to current work in progress.

**Comment by toddgauntgs** [comment]
@wesmc7777 Thanks for the info. While we're here, I'm not exactly sure how to install and use the provisioning_client headers. I have tried installing the library through apt-get, and have built and installed the azure-iot-sdk-c manually but I don't see any of the provisioning_client headers installed. Is this feature available yet or am I missing a build option?

Thanks
